github/connection_config.go

```diff
@@ -5,8 +5,11 @@ import (
 )
 
 type githubConfig struct {
-	Token   *string `hcl:"token"`
-	BaseURL *string `hcl:"base_url"`
+	Token          *string `hcl:"token,optional"`
+	BaseURL        *string `hcl:"base_url,optional"`
```

Comment:
Should `BaseURL` be optional?

Reply:
Yes, the `BaseURL` should be an optional parameter. This is only required for the GitHub Enterprise account.

Reply:
@ParthaI I think all the config arguments should remove the `optional` part.